### PR TITLE
plan: support DOT explain format

### PR DESCRIPTION
**Pull request by zz-jason**
usage example:

```sql
create table t(a bigint, b bigint);
desc format = "dot" select A.a, B.b from t A join t B on A.a > B.b where A.a < 10;
```
which returns:
```
TiDB > desc format = "dot" select A.a, B.b from t A join t B on A.a > B.b where A.a < 10;desc format = "dot" select A.a, B.b from t A join t B on A.a > B.b where A.a < 10;
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| dot contents                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|
digraph HashRightJoin_7 {
subgraph cluster7{
node [style=filled, color=lightgrey]
color=black
label = "root"
"HashRightJoin_7" -> "TableReader_10"
"HashRightJoin_7" -> "TableReader_12"
}
subgraph cluster9{
node [style=filled, color=lightgrey]
color=black
label = "cop"
"Selection_9" -> "TableScan_8"
}
subgraph cluster11{
node [style=filled, color=lightgrey]
color=black
label = "cop"
"TableScan_11"
}
"TableReader_10" -> "Selection_9"
"TableReader_12" -> "TableScan_11"
}
 |
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

copy and paste the dot file content to [http://www.webgraphviz.com/](http://www.webgraphviz.com/), that will generate a graph:

<img width="484" alt="2017-09-19 13 23 37" src="https://user-images.githubusercontent.com/5268763/30576941-cbf802a2-9d3d-11e7-955c-58300d4ee3d5.png">



if you have `dot` installed on your computer, you can do that by:
```sh
dot xx.dot -T png -O
```

### CODE REVIEW

**Comment by coocood** [comment]
Nice format!

File: plan/plans.go [hunk 1]
```diff
@@ -258,3 +260,58 @@ func (e *Explain) prepareRootTaskInfo(p PhysicalPlan) {
...
+	buffer := bytes.NewBufferString("")
+	buffer.WriteString(fmt.Sprintf("\ndigraph %s {\n", p.ExplainID()))
+	e.prepareeTaskDot(p, "root", buffer)
+	buffer.WriteString(fmt.Sprintln("}"))
+
+	row := types.MakeDatums(buffer.String())
+	e.Rows = append(e.Rows, row)
+}
+
+func (e *Explain) prepareeTaskDot(p PhysicalPlan, taskTp string, buffer *bytes.Buffer) {
```
Comment (coocood) [inline plan/plans.go]: s/prepareeTaskDot/prepareTaskDot

**Comment by shenli** [comment]
"explain format = "dot" select xxx " has the same result with "desc format = "dot" select xxx "

**Comment by zz-jason** [comment]
yes, `explain`, `desc`, `describe` are synonyms

**Comment by coocood** [comment]
LGTM

**Comment by XuHuaiyu** [comment]
easy to add test case?

**Comment by zz-jason** [comment]
to #4279

**Pull request merged**